webapp/src/js/bootstrapper/index.js

```diff
@@ -164,17 +176,21 @@
             });
         }
       })
-      .then(() => purger(localDb, userCtx, isInitialReplicationNeeded)
-        .on('start', () => setUiStatus('PURGE_INIT'))
-        .on('progress', function(progress) {
-          setUiStatus('PURGE_INFO', {
-            count: progress.purged,
-            percent: Math.floor((progress.processed / progress.total) * 100)
+      .then(() => {
+        return serverSidePurge
+          .shouldPurge(localDb, userCtx)
+          .then(shouldPurge => {
+            if (!shouldPurge) {
+              return;
+            }
+
+            return serverSidePurge
+              .purge(localDb, userCtx)
+              .on('start', () => setUiStatus('PURGE_INIT'))
+              .on('progress', progress => setUiStatus('PURGE_INFO', { count: progress.purged }))
+              .catch(console.error);
```

Comment:
Add a string prefix to make it easier to find where the error was thrown after it's minified, eg:
`.catch(err => console.error('Error attempting to purge', err));`

Reply:
Done.